model/src/main/java/com/javahelp/model/user/User.java

```diff
@@ -0,0 +1,84 @@
+package com.javahelp.model.user;
+import com.javahelp.model.user.UserInfo;
+import com.javahelp.model.user.ClientUserInfo;
+import com.javahelp.model.user.UserType;
+import com.javahelp.model.user.ProviderUserInfo;
+
+/**
+ * A class for a user entity
+ */
+public class User {
+
+    /**
+     * --- Attributes---
+     * stringID - the ID of the user account
+     * userInfo - contains all information of the user. It varies based on Client/Provider
+     */
+    private String stringID;
+    private UserInfo userInfo;
+    private String username;
+
+    /**
+     * A constructor for the current User Entity
+     *
+     * @param stringID - The ID of the account
+     * @param userInfo - The account information
+     * @param username - The username of the account (randomly generated)
+     */
+    public User(String stringID, UserInfo userInfo, String username) {
+        this.stringID = stringID;
+        this.userInfo = userInfo;
+        this.username = username;
+    }
+
+    /**
+     * Set the StringID for the user
+     *
+     * @param stringID - an updated ID of the account
+     */
+    public void setStringID(String stringID) {
+        this.stringID = stringID;
+    }
+
+    /**
+     * Set the StringID for the user
+     *
+     * @param userInfo - an updated ID of the account
+     */
+    public void setUserInfo(UserInfo userInfo) { this.userInfo = userInfo; }
+
+    /**
+     * Set the username for the user
+     *
+     * @param username - an updated username of the account
```

Comment:
You might want to check [this](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html) out, just to get an idea of how Javadocs are meant to look